### PR TITLE
fix(select): add background to selected item and remove color's prop

### LINKED ISSUE #561: [Fix][select] change background color on item selected
[thread info](https://juntossomosmais.slack.com/archives/C0117K4MYSW/p1725397393025589?thread_ts=1725311489.668669&cid=C0117K4MYSW)

### PR DESCRIPTION
**Pull request by felipefialho**
#### What is being delivered?

- Add a background to the selected item
- Improve hover color
- Remove the prop to choose the color of select

#### What impacts?

Improve colors after selecting an option
closes #561

#### Evidences

image

image

### CODE REVIEW

**Comment by tassioFront** [comment]
question: hey, man. It looks there is a background when there is a selected one. is that right?

[Screencast from 06-09-2024 16:50:14.webm](https://github.com/user-attachments/assets/5d4771b5-c490-44db-a528-e5a539f6f8e3)

File: packages/core/src/components.d.ts [hunk 1]
```diff
@@ -169,7 +169,6 @@ export namespace Components {
         "trigger"?: string;
     }
     interface AtomSelect {
-        "color"?: 'primary' | 'secondary' | 'danger';
```
Comment (tassioFront) [inline packages/core/src/components.d.ts:172]: perfect. I does not make sense

**Review by igorwessel — APPROVED** [review]

File: packages/core/src/components.d.ts (hunk 1 above)
Comment (felipefialho) [reply packages/core/src/components.d.ts:172]: This was left behind by Poc and we never forget to review it hahaha

Comment (tassioFront) [reply packages/core/src/components.d.ts:172]: Poc's copying and pasting issues haha

**Review by tassioFront — APPROVED** [review]

**Review by anabrook — APPROVED** [review]

**Review by MayaraRMA — APPROVED** [review]

**Pull request merged**